COM-135 feat: Adding Radio button component

cristiannietodev91 opened the pull request:
**Checklist**

- [X] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** 
[https://impulsumvc.atlassian.net/browse/COM-135](https://impulsumvc.atlassian.net/browse/COM-135)

cristiannietodev91 commented:
regarding this comment https://github.com/cmpsr/composer/pull/312#discussion_r788521065. The focus effect that are you talking about is achieved with this line in chakra-ui [Radio effect](https://github.com/chakra-ui/chakra-ui/blob/6e1bbf7bf261cf49a0c00349d63f62b855807ba7/packages/theme/src/components/checkbox.ts#L51). I removed that effect in this line [Line PR](https://github.com/cmpsr/composer/pull/321/files#diff-28d54516cc4e15c667076f50674503124f239a74be839db5c11242af6a0ca16eR24) because was not implemented in Figma. We might remove that line in order to have the effect again but the figma doesn't specify the boxShadow effect that we have to follow.

tkanzakic commented:
@cristiannietodev91 to change the base branch you only need to go to the title of the PR, click the Edit button and change the base branch, there was no need to create a new PR and lost all the comments in the previous one

cristiannietodev91 commented:
> @tkanzakic regarding this comment [#312 (comment)](https://github.com/cmpsr/composer/pull/312#discussion_r788521065). The focus effect that are you talking about is achieved with this line in chakra-ui [Radio effect](https://github.com/chakra-ui/chakra-ui/blob/6e1bbf7bf261cf49a0c00349d63f62b855807ba7/packages/theme/src/components/checkbox.ts#L51). I removed that effect in this line [Line PR](https://github.com/cmpsr/composer/pull/321/files#diff-28d54516cc4e15c667076f50674503124f239a74be839db5c11242af6a0ca16eR24) because was not implemented in Figma. We might remove that line in order to have the effect again but the figma doesn't specify the boxShadow effect that we have to follow.

I removed the `boxShadow: 'none',`  property and it looks more like Radio chakra UI

cristiannietodev91 commented:
> @cristiannietodev91 to change the base branch you only need to go to the title of the PR, click the Edit button and change the base branch, there was no need to create a new PR and lost all the comments in the previous one

@tkanzakic I tried to do it in that way and even rebasing locally and there are many conflicts. I did it to test again in the [previous PR](https://github.com/cmpsr/composer/pull/312) and it founds 500 files changed

tkanzakic reviewed (changes requested):
as mentioned [in the previous PR](https://github.com/cmpsr/composer/pull/312#pullrequestreview-856525801) the UI of the component do not match the designs in figma

cristiannietodev91 commented:
> as mentioned [in the previous PR](https://github.com/cmpsr/composer/pull/312#pullrequestreview-856525801) the UI of the component do not match the designs in figma

Can you add more details, please? I checked Figma twice if you add more details I can go through that

tkanzakic commented:
> > as mentioned [in the previous PR](https://github.com/cmpsr/composer/pull/312#pullrequestreview-856525801) the UI of the component do not match the designs in figma
> 
> Can you add more details, please? I checked Figma twice if you add more details I can go through that.
> 
> I added two additional modification regarding the `focus` effect. apart from that I don't see what in the ticket is not align with the figma

if you inspect your component you will see all the difference, just to mention a couple of them, font color and font sizes are not the right ones

guilean commented on packages/components/src/components/primitives/Radio/types.ts:
```diff
@@ -0,0 +1,13 @@
+  'default',
+] as const;
+export type RadioVariant = typeof radioVariants[number];
+export const RadioSizes = ['sm', 'md', 'lg'] as const;
+export type RadioSize = typeof RadioSizes[number];
```
Sizes according to design are `s`, `m` and `l`, being `l` the default size.

cristiannietodev91 replied:
There is not a clear guideline to follow in this case for example within the `Button` component the sizes defined are `xs`, `s` , `m` and `l`, and the current sizes defined for the button component are similar to the ones I defined in the Radio component.
Can you please share with me the right guideline?  a new ticket must be created to organize `Button` component?. 
@guilean

guilean replied:
Let's stick to the ones defined in figma if @tkanzakic agrees on that

tkanzakic replied:
the button is not consistent with figma because the design team updated the figma file after the component was implemented, we have to update the button to match figma, figma is the source of truth

guilean commented on packages/components/src/theme/ComposerTheme/Components/Radio.ts:
```diff
@@ -0,0 +1,71 @@
+import { ComponentMultiStyleConfig } from '@chakra-ui/theme';
+import { PartsStyleInterpolation } from '@chakra-ui/theme-tools';
+
+const sizes: Record<string, PartsStyleInterpolation<typeof parts>> = {
```
Update sizes to match design

cristiannietodev91 replied:
The same discussion here. https://github.com/cmpsr/composer/pull/321/files#r792578077

guilean commented on packages/components/src/theme/ComposerTheme/Components/Radio.ts:
```diff
@@ -0,0 +1,71 @@
+};
+
+export const Radio: ComponentMultiStyleConfig = {
+  parts: parts.keys,
```
Is this needed?

cristiannietodev91 replied:
yes. it was necessary because I was using `ComponentMultiStyleConfig` type that requires parts prop but it existed a union type `ComponentStyleConfig` which join `ComponentMultiStyleConfig` and `ComponentSingleStyleConfig` and in that case that field is not needed. 

just to add that following ChakraUi documentation using parts props is recommended when you are working with [MultiPart components ](https://chakra-ui.com/docs/theming/component-style#styling-multipart-components) I think it makes more readable the component.

guilean commented on packages/components/src/theme/ComposerTheme/Components/Radio.ts:
```diff
@@ -0,0 +1,71 @@
+      _checked: {
+        borderColor: 'primary-default',
+        bg: 'primary-default',
+        color: 'text-light',
```
Is this needed? Isn't it enough to have it defined in the color of the control part?

cristiannietodev91 replied:
it is needed because a color value is defined in `Checkbox` and I need to override it

guilean reviewed (changes requested):
Please create a changeset 🙏🏻

cristiannietodev91 commented:
> > > as mentioned [in the previous PR](https://github.com/cmpsr/composer/pull/312#pullrequestreview-856525801) the UI of the component do not match the designs in figma
> > 
> > 
> > Can you add more details, please? I checked Figma twice if you add more details I can go through that.
> > I added two additional modification regarding the `focus` effect. apart from that I don't see what in the ticket is not align with the figma
> 
> if you inspect your component you will see all the difference, just to mention a couple of them, font color and font sizes are not the right ones

Yes, you are right. I didn't apply the font style to the right section of the radio component I just updated the PR.
@tkanzakic can you check again.

tkanzakic reviewed (changes requested):
- missing `index.ts` file at the component level that re-export the component and its props
- missing re-export in the parent (primitives) `index.ts`
